framework/core/src/Database/Exception/MigrationKeyMissing.php

```diff
@@ -13,13 +13,13 @@
 
 class MigrationKeyMissing extends Exception
 {
-    public function __construct(protected string $direction, string $file = null)
+    public function __construct(protected string $direction, ?string $file = null)
     {
         $fileNameWithSpace = $file ? ' '.realpath($file) : '';
         parent::__construct("Migration file $fileNameWithSpace should contain an array with up/down (looking for $direction)");
     }
 
-    public function withFile(string $file = null): self
+    public function withFile(?string $file = null): self
```

Comment:
That file is optional seems to be counter intuitive? One could just not call construct with the 2nd argument or we could introduce a `withoutFile` method instead?

Reply:
@luceos there is no difference in behavior, it was optional before, it is still optional now, the typing is just more clear that it can be null

before you could call ->withFile() with no parameter, but couldn't call ->withFile(null), now you can call ->withFile(null) because it should be allowed if you can call ->withFile() and the arg is assumed null

this applies to all your comments 👍🏼

Reply:
Yeah, I completely understand the logic the method offers. I'm talking about the method being named "with file" and the method allowing a null value, implying there is no file. Semantically, the logic of the method goes against the name of the method.

Reply:
@luceos I understand, but that's not really a semantic added by this PR anyway, it's there before and after this PR, so it's out of scope for the purpose of the changes here, it would be best to open an issue for new stuff, just we can stay focused on one thing at a time imo 🫡

Reply:
I've pushed a commit to make the argument required since it's only used in one spot that provides a string